zlib-rs/src/inflate/window.rs

```diff
@@ -195,68 +184,82 @@ unsafe fn slice_assume_init(slice: &[MaybeUninit<u8>]) -> &[u8] {
 mod test {
     use super::*;
 
+    use crate::allocate::Allocator;
+
+    fn init_window(window_bits_log2: usize) -> Window<'static> {
+        let mut window = Window::new_in(&Allocator::RUST, window_bits_log2).unwrap();
+        window.buf.fill(MaybeUninit::new(0));
+        window.have = 0;
+        window.next = 0;
+        window
+    }
+
     #[test]
     fn extend_in_bounds() {
         let mut checksum = 0;
 
-        let mut buf = [0; 12];
-        let mut window = Window::from_slice(&mut buf);
+        let mut window = init_window(4);
```

Comment:
we now actually allocate the window, so these tests needed to be slightly adjusted because a window is now always some power of 2 bytes